interface/ApfpBlas.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <gmp.h> 
+#include <functional>
+#include "ApfpInterfaceType.h"
+
+// 
+using IndexFunction = std::function<ApfpInterfaceTypePtr(unsigned long)>;
+using ConstIndexFunction = std::function<ApfpInterfaceTypeConstPtr(unsigned long)>;
+
+
+/// Null terminated string describing the most recent library error if available
+/// Pointer is only guaranteed to live until the next library call
+const char* ApfpErrorDescription();
+
+int ApfpInit(unsigned long precision);
```

Comment:
Why do we want this C-like interface? Isn't the object-based one enough for our purposes?

Reply:
The BLAS interface drops into SDPB (the diff is like 3 lines), and the OO interface introduces a bunch of transitive dependencies in SDPB. The middle ground is sticking to the BLAS interface, so we only have to link against APFP.

Reply:
Here's the diff inside SDPB. It's very small

https://github.com/ChrisPattison/sdpb/compare/master...fpga#diff-400740df4c03d1d93480e9a380e53e6ec68ffe6fcc16f61977e2d018a3a73499

Reply:
The C style interface allows you to avoid having to plumb through objects on the SDPB side. for the most part APFP looks like any other BLAS library which is a feature in this case

Reply:
Okay sure, works for me. Why does the OO interface introduce more dependencies than BLAS?

Reply:
The objects are holding hlslib handles and I can't think of a good way to get rid of that. I was fairly careful in the ApfpBlas header to avoid including anything from our dependencies besides MPFR/GMP. We get around it by holding a global singleton of `Apfp` in `ApfpBlas.cpp`